docs/content/docs/connectors/datastream/prometheus.md

```diff
@@ -0,0 +1,473 @@
+
+{{< connector_artifact flink-connector-prometheus prometheus >}}
+
+## Usage
+
+The Prometheus sink provides a builder class to build a `PrometheusSink` instance. The code snippets below shows 
+how to build a `PrometheusSink` with a basic configuration, and an optional [request signer](#request-signer).
+
+```java
+PrometheusSink sink = PrometheusSink.builder()
+        .setPrometheusRemoteWriteUrl(prometheusRemoteWriteUrl)
+        .setRequestSigner(new AmazonManagedPrometheusWriteRequestSigner(prometheusRemoteWriteUrl, prometheusRegion)) // Optional
+        .build();
+```
+The only **required** configuration is `prometheusRemoteWriteUrl`. All other configurations are optional.
+
+If your sink has parallelism > 1, you need to ensure the stream is keyed using the `PrometheusTimeSeriesLabelsAndMetricNameKeySelector` 
+key selector, so that all samples of the same time-series are in the same partition and order is not lost.
+See [Sink parallelism and keyed streams](#sink-parallelism-and-keyed-streams) for more details.
+
+
+### Input data objects
+
+The sink expects `PrometheusTimeSeries` records as input.
+Your input data must be converted into `PrometheusTimeSeries`, using a map or flatMap operator, before the sending to the sink.
+
+`PrometheusTimeSeries` instances are immutable and cannot be reused. You can use the [builder](#populating-a-prometheustimeseries)
+to create and populate instances.
+
+A `PrometheusTimeSeries` represents a single time-series record when sent to the Remote Write interface. Each time-series 
+record may contain multiple samples.
+
+{{< hint info >}}
+In the context of Prometheus, the term "time-series" is overloaded.
+It means both *a series of samples with a unique set of labels* (a time-series in the underlying time-series database),
+and *a record sent to the Remote Write interface*. A `PrometheusTimeSeries` instance represents a record sent to the interface.
+
+The two concepts are related, because time-series "records" with the same sets of labels are sent to the same
+"database time-series".{{< /hint >}}
+
+Each `PrometheusTimeSeries` record contains:
+
+- One **`metricName`**. A string that is translated into the value of the `__name__` label.
+- Zero or more **`Label`** entries. Each label has a `key` and a `value`, both `String`. Labels represent additional dimensions of the samples. Duplicate Label keys are not allowed.
+- One or more **`Sample`**. Each sample has a `value` (`double`) representing the measure, and a `timestamp` (`long`) representing the time of the measure, in milliseconds from the Epoch. Duplicate timestamps in the same record are not allowed.
+
+The following pseudocode represents the structure of a `PrometheusTimeSeries` record:
+
+```
+PrometheusTimeSeries
+  + --> (1) metricName <String>
+  + --> (0..*) Label
+            + name <String>
+            + value <String>
+  + --> 1..* Sample
+            + timestamp <long>
+            + value <double>   
+```
```

Comment:
The numbers are a little confusing - can we instead add comments on the right to explain optional / non-optional?

Reply:
Really? 
This is supposed to be pseudo-code and uses the standard cardinality notation commonly used in diagrams, like ER or UML.

Labels and Samples are lists, is not a matter of being mandatory. To convey the fact one list may contain zero or more elements and the other must contain at least one element, it would become something like this, IMO this is more confusing:

```
  + --> metricName <String> // mandatory
  + --> list of Label // zero or more elements
            + name <String>
            + value <String>
  + --> list of Sample // one or more elements
            + timestamp <long>
            + value <double>  
```


I think in the original there was a parenthesis missing. Fixing it it would be this:
```
  + --> (1) metricName <String>
  + --> (0..*) Label
            + name <String>
            + value <String>
  + --> (1..*) Sample
            + timestamp <long>
            + value <double>   
 ```  
 
 alternatively this one
 
 ```
  + --> (mandatory) metricName <String>
  + --> (list of zero or more) Label
            + name <String>
            + value <String>
  + --> (list of one or more) Sample
            + timestamp <long>
            + value <double>   
 ```